packages/atlas/src/views/studio/VideoWorkspace/VideoWorkspace.tsx

```diff
@@ -1,30 +1,34 @@
-import React, { useCallback, useEffect, useState } from 'react'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
 import { CSSTransition, SwitchTransition } from 'react-transition-group'
-import useMeasure from 'react-use-measure'
 
 import { ActionBarVariant } from '@/components/ActionBar'
 import { DrawerHeader } from '@/components/DrawerHeader'
 import { useDisplayDataLostWarning } from '@/hooks/useDisplayDataLostWarning'
 import { useHeadTags } from '@/hooks/useHeadTags'
 import { useMediaMatch } from '@/hooks/useMediaMatch'
+import { NftAuctionInputMetadata, NftIssuanceInputMetadata } from '@/joystream-lib'
 import {
   VideoFormData,
   VideoWorkspaceFormStatus,
+  VideoWorkspaceVideoFormFields,
   useVideoWorkspace,
   useVideoWorkspaceData,
 } from '@/providers/videoWorkspace'
 import { cVar, transitions } from '@/styles'
+import { Listing } from '@/views/studio/VideoWorkspace/NFTForm/types'
 
 import { NFTForm } from './NFTForm'
 import { VideoForm } from './VideoForm'
 import { Container, DrawerOverlay, ScrollContainer, StyledActionBar } from './VideoWorkspace.style'
 import { useHandleVideoWorkspaceSubmit } from './hooks'
 
 export const VideoWorkspace: React.FC = React.memo(() => {
-  const [formStatus, setFormStatus] = useState<VideoWorkspaceFormStatus | null>(null)
+  const [listingType, setListingType] = useState<Listing>(undefined)
+  const [formStatus, setFormStatus] = useState<VideoWorkspaceFormStatus<VideoWorkspaceVideoFormFields> | null>(null)
+  const [NFTFormStatus, setNFTFormStatus] = useState<VideoWorkspaceFormStatus<NftAuctionInputMetadata> | null>(null)
```

Comment:
Why do we need those to be separate, couldn't we hold just a single form status?